public/dorequest.php

```diff
@@ -229,17 +230,26 @@ function DoRequestError(string $error, ?int $status = 200, ?string $code = null)
 
     case "gameslist":
         $consoleID = (int) request()->input('c', 0);
-        $response['Response'] = getGamesListDataNamesOnly($consoleID);
+        $cacheKey = 'connect:gameslist:' . $consoleID;
+        $response['Response'] = Cache::remember($cacheKey, now()->addMinutes(5), function () use ($consoleID) {
+            return getGamesListDataNamesOnly($consoleID);
+        });
         break;
 
     case "officialgameslist":
         $consoleID = (int) request()->input('c', 0);
-        $response['Response'] = getGamesListDataNamesOnly($consoleID, true);
+        $cacheKey = 'connect:officialgameslist:' . $consoleID;
+        $response['Response'] = Cache::remember($cacheKey, now()->addMinutes(5), function () use ($consoleID) {
+            return getGamesListDataNamesOnly($consoleID, true);
+        });
         break;
 
     case "hashlibrary":
         $consoleID = (int) request()->input('c', 0);
-        $response['MD5List'] = getMD5List($consoleID);
+        $cacheKey = 'connect:hashlibrary:' . $consoleID;
+        $response['MD5List'] = Cache::remember($cacheKey, now()->addMinutes(5), function () use ($consoleID) {
+            return getMD5List($consoleID);
+        });
```

Comment:
I don't think this will gain us much. There were only a total of 633 requests (of 342k) to these endpoints yesterday.
```
forge@app.retroachievements.org /var/log/nginx $ grep r=gameslist connect.retroachievements.org-access.log | wc -l
0
forge@app.retroachievements.org /var/log/nginx $ grep r=officialgameslist connect.retroachievements.org-access.log | wc -l
224
forge@app.retroachievements.org /var/log/nginx $ grep r=hashlibrary connect.retroachievements.org-access.log | wc -l
409
```

And 880 (of 649k) the day before.

Reply:
The first two functions could be hugely improved by just querying the ID/Title instead of doing a full data dump and extracting the Title.